Base/QTGUI/qSlicerApplication.cxx

```diff
@@ -396,21 +396,24 @@ bool qSlicerApplication::notify(QObject *receiver, QEvent *event)
   catch( std::exception& exception )
     {
     QString errorMessage;
-    errorMessage = tr("%1 has caught an internal error.\n\n").arg(this->applicationName());
-    errorMessage += tr("You may be able to continue from this point, but results are undefined.\n\n");
-    errorMessage += tr("Suggested action is to save your work and restart.");
-    errorMessage += tr("\n\nIf you have a repeatable sequence of steps that causes this message, ");
+    errorMessage = tr("%1 has caught an exception, ").arg(this->applicationName());
+    errorMessage += tr("please save your work and restart.\n\n");
+    errorMessage += tr("If you have a repeatable sequence of steps that causes this message, ");
     errorMessage += tr("please report the issue following instructions available at http://slicer.org\n\n\n");
     errorMessage += tr("The message detail is:\n\n");
     errorMessage += tr("Exception thrown in event: ") + exception.what();
+    if (!tr("bad allocation").compare(exception.what()) || !tr("std::bad_alloc").compare(exception.what()))
+      {
+        errorMessage += tr("\n\nIncreasing system memory may resolve this issue\n\n");
```

Comment:
Most users stop reading after a couple of words, so it would be better to start with this specific message and add the general stuff after that.

I think we could also improve the memory allocation message:
"The application has run out of memory." and then
- For Windows "Increasing virtual memory size in system settings or adding more RAM may fix this issue".
- For Linux "Increasing swap size in system settings or adding more RAM may fix this issue".
- For Mac "Increasing free disk space or adding more RAM may fix this issue".

Reply:
System kernels are checked before giving the additional suggestion.